packages/core/admin/admin/src/content-manager/pages/ListView/components/BulkActionButtons/index.js

```diff
@@ -71,14 +79,68 @@ const confirmDialogsPropTypes = {
 
 const ConfirmDialogPublishAll = ({ isOpen, onToggleDialog, isConfirmButtonLoading, onConfirm }) => {
   const { formatMessage } = useIntl();
+  const { get } = useFetchClient();
+  const { selectedEntries } = useTableContext();
+  const toggleNotification = useNotification();
+  const { formatAPIError } = useAPIErrorHandler(getTrad);
+  const {
+    contentType: { uid: slug },
+  } = useSelector(listViewDomain());
+
+  const {
+    data: countDraftRelations,
+    isLoading,
+    isError,
+    error,
+  } = useQuery(
+    ['content-manager', 'draft-relations'],
+    async () => {
+      /**
+       * TODO: refactor the API call when the param serializer will be ready
+       */
+      const {
+        data: { data },
+      } = await get(
+        `/content-manager/collection-types/${slug}/actions/countManyEntriesDraftRelations?${stringify(
+          { ids: selectedEntries },
+          { encode: false }
+        )}`
+      );
+
+      return data;
+    },
+    {
+      enabled: isOpen,
+    }
+  );
+
+  if (isError) {
+    toggleNotification({ type: 'warning', message: formatAPIError(error) });
+
+    return null;
+  }
 
   return (
     <ConfirmBulkActionDialog
-      isOpen={isOpen}
+      isOpen={isOpen && !isLoading && !isError}
```

Comment:
Are you sure you want to show the overlay only after loading has finished? Doesn't that mean there could be a delay between a user action and the modal opening?

Reply:
I did it because otherwise, if we have an error in the API we show the Confirmation overlay before the API response and then we hide the Confirmation overlay and we show the Error (with the conditions instead I can show the Confirmation overlay only if the API call has no errors, the same behaviour that we have when we publish an entry from its edit view)